packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/Axes/index.tsx

```diff
@@ -102,6 +107,35 @@ export const Axes: FC<Props> = ({ itemsMap }) => {
                             setMax={(newValue) => setXMaxValue(0, newValue)}
                         />
                     )}
+
+                    {isNumericItem(xAxisField) && !dirtyLayout?.flipAxes && (
+                        <>
+                            <Switch
+                                label="Truncate x-axis"
+                                checked={
+                                    dirtyEchartsConfig?.xAxis?.[0]
+                                        ?.minOffset !== undefined ||
+                                    dirtyEchartsConfig?.xAxis?.[0]
+                                        ?.maxOffset !== undefined
+                                }
+                                onChange={(e) => {
+                                    if (e.target.checked) {
+                                        setXMaxOffsetValue(
+                                            0,
+                                            DEFAULT_OFFSET_VALUE_FOR_MANUAL_RANGE_PERCENTAGE,
```

Comment:
this is set here, to add it them to the chart config. I'm happy to change it to be set to _true_ or similar and use this constant in `useCartesianChartConfig`